src/set/set.service.ts

```diff
@@ -1,345 +1,418 @@
+    }
 
-    if (!set)
-      throw new NotFoundException()
+    async getOneSet(id: ObjectId): Promise<ResponseSetWithTasks> {
+        const set: ResponseSet & { tasks: ResponseTaskWithStatus[] } =
+            await this.setSchema
+                .findOne(
+                    { _id: id, status: Status.ACTIVE },
+                    {
+                        _id: 1,
+                        daresCount: 1,
+                        truthCount: 1,
+                        name: 1,
+                        language: 1,
+                        createdBy: 1,
+                        tasks: 1,
+                        previewImage: 1,
+                        bannerImage: 1,
+                    },
+                )
+                .populate<ResponseSet & { tasks: ResponseTaskWithStatus[] }>(
+                    'createdBy',
+                    '_id username',
+                )
+                .lean()
+
+        if (!set) throw new NotFoundException()
+
+        // Remove tasks from array that are not active
+        const result: ResponseSetWithTasks = this.onlyActiveTasks(set)
+
+        return result
+    }
 
-    // Remove tasks from array that are not active
-    const result: ResponseSetWithTasks = this.onlyActiveTasks(set)
+    async updateSetMetadata(
+        id: ObjectId,
+        updateSetDto: UpdateSetDto,
+        user: JwtUserDto,
+    ): Promise<ResponseSetMetadata> {
+        let queryMatch: { _id: ObjectId; createdBy?: ObjectId } = { _id: id }
 
-    return result;
-  }
+        if (user.role !== Role.Admin) {
+            queryMatch.createdBy = user.userId
+        }
 
-  async updateSetMetadata(id: ObjectId, updateSetDto: UpdateSetDto, user: JwtUserDto): Promise<ResponseSetMetadata> {
+        const set: ResponseSetMetadata = await this.setSchema.findOneAndUpdate(
+            queryMatch,
+            updateSetDto,
+            {
+                new: true,
+                select: '_id daresCount truthCount language name createdBy',
+            },
+        )
 
-    let queryMatch: { _id: ObjectId, createdBy?: ObjectId } = { _id: id }
+        if (!set) throw new NotFoundException()
 
-    if (user.role !== Role.Admin) {
-      queryMatch.createdBy = user.userId
+        return set
     }
 
-    const set: ResponseSetMetadata = await this.setSchema.findOneAndUpdate(queryMatch, updateSetDto, { new: true, select: "_id daresCount truthCount language name createdBy" })
-
-    if (!set)
-      throw new NotFoundException()
+    async deleteSet(
+        id: ObjectId,
+        deleteType: string,
+        user: JwtUserDto,
+    ): Promise<void> {
+        // Hard delete
+        if (deleteType === 'hard') {
+            if (user.role != 'admin') throw new ForbiddenException()
 
-    return set
+            const set = await this.setSchema.findByIdAndDelete(id)
 
-  }
+            if (!set) throw new NotFoundException()
 
-  async deleteSet(id: ObjectId, deleteType: string, user: JwtUserDto): Promise<void> {
+            return
```

Comment:
while there is no rule written down anywhere the user endpoints return the deleted object....so maybe we want to find a way to unify this

Reply:
Best practice is 204